src/inspect_ai/_view/server.py

```diff
@@ -457,14 +489,73 @@ async def list_eval_logs_async(
     fs = filesystem(log_dir, fs_options)
     if fs.is_async():
         async with async_fileystem(log_dir, fs_options=fs_options) as async_fs:
-            if await async_fs._exists(log_dir):
+            # Attempt existence check with robust handling for Azure-style auth issues.
+            try:
+                exists = await async_fs._exists(log_dir)
+            except Exception as ex:  # noqa: BLE001
+                # For remote filesystems we may encounter authentication errors for
+                # existence probes even when blob listing might succeed (or we want
+                # to provide a clearer diagnostic). Detect common auth failures and
+                # return an empty listing with a helpful warning rather than a 500.
+                protocol, _ = split_protocol(log_dir)
+                scheme = (protocol or "file").lower()
+                ex_msg = str(ex)
+                if scheme in {"az", "abfs", "abfss"} and (
+                    "authenticate" in ex_msg
+                    or "NoAuthenticationInformation" in ex_msg
+                    or "AuthenticationFailed" in ex_msg
+                ):
+                    logger.warning(
+                        "Azure storage authentication failed while probing '%s'. "
+                        "Suppressed stack trace. Guidance: (a) run 'az login' or ensure role "
+                        "assignment (Storage Blob Data Reader/Contributor); (b) if using SAS set "
+                        "AZURE_STORAGE_SAS_TOKEN (and AZURE_STORAGE_ACCOUNT_NAME if needed); (c) if using account key, "
+                        "set AZURE_STORAGE_ACCOUNT_KEY. Original error: %s",
+                        log_dir,
+                        ex_msg,
+                    )
+                    return []
+                # TODO: Add S3 login error catching, as well as any other remote file system of interest
+
+                # Re-raise non-auth related issues
+                raise
+
+            if exists:
                 # prevent caching of listings
                 async_fs.invalidate_cache(log_dir)
                 # list logs
                 if recursive:
                     files: list[dict[str, Any]] = []
-                    async for _, _, filenames in async_fs._walk(log_dir, detail=True):
-                        files.extend(filenames.values())
+                    try:
```

Comment:
I think maybe we want to actually detect that we are using a provider that we know can't handle detail and just enter that codepath expicitly rather than relying on a type error. Alternatively, I'd be fine just interrogating the  `_walk` method to see if it takes a `detail` parameter.